pkg/network/networknamescheme/networknamescheme.go

```diff
@@ -19,4 +19,35 @@
 
 package networknamescheme
 
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
 const DefaultNetworkName = "eth0"
+
+func GetNetworkNameSchemeMap(vmi *v1.VirtualMachineInstance) map[string]string {
+	networkNameSchemeMap := map[string]string{}
+
+	mapDefaultNetworkToPodInterfaceName(vmi, networkNameSchemeMap)
+	mapMultusNonDefaultNetworksToPodInterfaceName(vmi, networkNameSchemeMap)
+	return networkNameSchemeMap
+}
+
+func mapMultusNonDefaultNetworksToPodInterfaceName(vmi *v1.VirtualMachineInstance, networkNameSchemeMap map[string]string) {
+	multusNonDefaultNetworkList := vmispec.FilterMultusNonDefaultNetworks(vmi.Spec.Networks)
+	for i, network := range multusNonDefaultNetworkList {
+		networkNameSchemeMap[network.Name] = fmt.Sprintf("net%d", i+1)
+	}
+}
+
+func mapDefaultNetworkToPodInterfaceName(vmi *v1.VirtualMachineInstance, networkNameSchemeMap map[string]string) {
```

Comment:
Same comment here, I would try to avoid mutation like this.
But for that, in this case, it will need to be done at the caller.
```
if multusDefaultNetwork, err := vmispec.LookUpDefaultNetwork(vmi.Spec.Networks); err == nil {
    networkNameSchemeMap[multusDefaultNetwork.Name] = DefaultNetworkName
}
```

An alternative is to return the mutated map.

Reply:
plus one on avoiding mutating the inner map.

Reply:
+1